src/OpenTelemetry.Api/Trace/TelemetrySpan.cs

```diff
@@ -303,6 +304,64 @@ public TelemetrySpan AddBaggage(string key, string value)
             return this;
         }
 
+        /// <summary>
+        /// Record Exception.
+        /// </summary>
+        /// <param name="ex">Exception to be recorded.</param>
+        /// <returns>The <see cref="TelemetrySpan"/> instance for chaining.</returns>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public TelemetrySpan RecordException(Exception ex)
+        {
+            Dictionary<string, object> attributes = new Dictionary<string, object>
+            {
+                { SemanticConventions.AttributeExceptionType, ex.GetType().Name },
+                { SemanticConventions.AttributeExceptionStacktrace, ex.ToInvariantString() },
```

Comment:
Due to the potential size of the stack trace, I feel that later we should have an option to not add it.

Reply:
it would be better to trim instead of not add?

Reply:
Do you mean an option in this method itself?
Or for individual exporters?

Reply:
Good questions.

It seems good to have "config" options to limit the size of attribute values in general: trim at some upper bound, and also an option to not add the stack trace. Both should be off by default: they are more like escape valves if this becomes a problem with actual usage.

I see these settings as part of API configuration. Anyway, I see it as separate from this PR.